src/Cassandra/Connections/Connection.cs

```diff
@@ -712,7 +723,7 @@ void DeserializeResponseStream(MemoryStream stream, long timestamp)
                         plainTextStream = compressor.Decompress(new WrappedStream(stream, header.BodyLength));
                         plainTextStream.Position = 0;
                     }
-                    response = FrameParser.Parse(new Frame(header, plainTextStream, serializer));
+                    response = FrameParser.Parse(new Frame(header, plainTextStream, serializer), resultMetadata);
```

Comment:
Result metadata could be stored at Frame level, that way we can avoid passing the result metadata as parameter for  `Create()` handlers, wdyt?

Reply:
I like that idea, I'll change it